src/pip/_internal/req/req_file.py

```diff
@@ -537,7 +538,8 @@ def get_file_content(url: str, session: PipSession) -> Tuple[str, str]:
 
     # Assume this is a bare path.
     try:
-        with open(url, "rb") as f:
+        # Resolve symlink'd directories
+        with open(os.path.abspath(url), "rb") as f:
```

Comment:
`abspath` famously does not resolve symlinks, so either the implementation is wrong or the comment needs further clarification.